dashboard/admin.php

```diff
@@ -1,276 +1,312 @@
+            case 'top':
+                $new_content = $player_holder['string'] . $content;
+                break;
+            case 'middle':
+                $middle_pos = round(sizeof($body) / 2);
+                for ($i = 0; $i < sizeof($body); $i++) {
 
-            if (!empty($params['show_video_title']) && $params['show_video_title'] !== null)
-                $dm_ce_data += ['show_video_title' => self::sanitize_this('show_video_title')];
+                        $new_content .= $dom->saveHTML($body[$i]);
 
-            if (!empty($params['show_info_card']) && $params['show_info_card'] !== null)
-                $dm_ce_data += ['show_info_card' => self::sanitize_this('show_info_card')];
+                        if ($i == $middle_pos - 1) {
+                            $new_content .= $player_holder['string'];
+                        }
+                }
+                break;
+            default:
+                $new_content = $content . $player_holder['string'];
+                endswitch;
 
-            if (!empty($params['show_carousel_playlist']) && $params['show_carousel_playlist'] !== null)
-                $dm_ce_data += ['show_carousel_playlist' => self::sanitize_this('show_carousel_playlist')];
+            } else {
+                $new_content = $content;
+            }
 
-            if (!empty($params['mute']) && $params['mute'] !== null)
-                $dm_ce_data += ['mute' => self::sanitize_this('mute')];
+            return $new_content;
 
-            // Save the option to database
-            update_option('dm_ce_options_' . $tab, $dm_ce_data);
+        } // if the post and the page
 
-            echo '<div id="setting-error-settings_updated" class="notice notice-success settings-error is-dismissible"> 
-                    <p><strong>' . esc_html( __('Settings saved', 'dm_embed_plugin') ) . '</strong></p>
-                    </div>';
-        }
+        return $content;
     }
 
-    private function store_credentials($params)
+    public function migrate_old_player_to_custom_embed($content)
     {
-        if (!empty($params) && wp_verify_nonce(self::sanitize_this('dm_save_data'), 'dm_save_data')) {
-
-            $dm_ce_data = [];
-
-            if (!empty($params['api_key']) && $params['api_key'] !== null)
-                $dm_ce_data += ['api_key' => self::sanitize_this('api_key')];
-
-            if (!empty($params['api_secret']) && $params['api_secret'] !== null)
-                $dm_ce_data += ['api_secret' => self::sanitize_this('api_secret')];
-
-            update_option('dm_ce_credentials', $dm_ce_data);
-
-            echo '<div id="setting-error-settings_updated" class="notice notice-success settings-error is-dismissible"> 
-                    <p><strong>' . esc_html( __('Settings saved', 'dm_embed_plugin') ) . '</strong></p>
-                    </div>';
-        }
-    }
+        $regexGetIframe = '(?:<iframe[^>]*)(?:(?:\/>)|(?:>.*?<\/iframe>))';
```

Comment:
@isatrio  we need to find all the iframes inside the article content using this regex